File: CSHARP/Cap08/c08ex09.cs
// c08ex09.cs

using System;
using static System.Console;

class Program
{
    static int Fibonacci(int N)
    {
        if (N == 0)
            return 0;
        if (N == 1)
            return 1;
        if (N >= 2)
            return Fibonacci(N - 1) + Fibonacci(N - 2);
        return 0; // Para evitar aviso de retorno implícito
    }

    static void Main(string[] args)
    {
        for (int I = 0; I <= 14; I++)
            WriteLine($"{I} - {Fibonacci(I)}");

        Write("\nPressione <Enter> para sair... ");
        ReadLine();
    }
}

File: VB_NET/Cap08/c08ex09.vb
' c08ex09.vb

Imports System
Imports System.Console

Module Program
    Function Fibonacci(ByVal N As Integer) As Integer
        If (N = 0) Then
            Return 0
        End If
        If (N = 1) Then
            Return 1
        End If
        If (N >= 2) Then
            Return Fibonacci(N - 1) + Fibonacci(N - 2)
        End If
        Return 0 ' Para evitar aviso de retorno implícito
    End Function

    Sub Main(args As String())
        For I As Integer = 0 To 14
            WriteLine($"{I} - {Fibonacci(I)}")
        Next
        Write(vbCrLf & "Pressione <Enter> para sair... ")
        ReadLine()
    End Sub
End Module
